src/Polly.Testing/ResilienceStrategyExtensions.cs

```diff
@@ -32,16 +27,23 @@ public static IReadOnlyList<ResilienceStrategyDescriptor> GetInnerStrategies<TRe
     /// <param name="strategy">The strategy instance.</param>
     /// <returns>A list of inner strategies.</returns>
     /// <exception cref="ArgumentNullException">Thrown when <paramref name="strategy"/> is <see langword="null"/>.</exception>
-    public static IReadOnlyList<ResilienceStrategyDescriptor> GetInnerStrategies(this ResilienceStrategy strategy)
+    public static InnerStrategiesDescriptor GetInnerStrategies(this ResilienceStrategy strategy)
     {
         Guard.NotNull(strategy);
 
         var strategies = new List<ResilienceStrategy>();
         strategy.ExpandStrategies(strategies);
 
-        return strategies.Select(s => new ResilienceStrategyDescriptor(s.Options, GetType(s), s.GetType())).ToList();
+        var innerStrategies = strategies.Select(s => new ResilienceStrategyDescriptor(s.Options, s.GetType())).ToList();
+
+        return new InnerStrategiesDescriptor(
+            innerStrategies.Where(s => !ShouldSkip(s.StrategyType)).ToList().AsReadOnly(),
```

Comment:
Do we need to force the list to be read-only?

Reply:
Yes, after the extension it should not be modifiable anymore.